File: lab3/client1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>
#include <sys/socket.h>

#define PORT 8080
#define SERVER_IP "127.0.0.1"
#define MAX_MSG_LEN 1024

int main() {
    int clientSocket;
    struct sockaddr_in serverAddr;

    // Create a socket
    clientSocket = socket(AF_INET, SOCK_STREAM, 0);
    if (clientSocket == -1) {
        perror("socket");
        exit(1);
    }

    serverAddr.sin_family = AF_INET;
    serverAddr.sin_port = htons(PORT);
    serverAddr.sin_addr.s_addr = inet_addr(SERVER_IP);

    // Connect to the server
    if (connect(clientSocket, (struct sockaddr*)&serverAddr, sizeof(serverAddr)) == -1) {
        perror("connect");
        exit(1);
    }

    char message[MAX_MSG_LEN];

    while (1) {
        printf("Client: ");
        fgets(message, sizeof(message), stdin);

        // Send the message to the server
        send(clientSocket, message, strlen(message), 0);

        // Receive a response from the server
        recv(clientSocket, message, sizeof(message), 0);
        printf("Server: %s", message);
    }

    close(clientSocket);

    return 0;
}

File: lab7/client1.py
import socket

# Define the server address and port
SERVER_ADDRESS = ('127.0.0.1', 8080)

# Create a socket and connect to the server
client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
client_socket.connect(SERVER_ADDRESS)

# Get user login credentials
username = input("Enter username: ")
password = input("Enter password: ")

# Send login credentials to the server
client_socket.send(username.encode())
client_socket.send(password.encode())

# Receive and print the server's response
response = client_socket.recv(1024).decode()
print(response)

if response == "Login Successful":
    while True:
        choice = int(input("Enter your choice (1-Debit, 2-Credit, 3-View Balance, 4-EXIT): "))
        client_socket.send(str(choice).encode())

        if choice == 1 or choice == 2:
            amount = float(input("Enter the amount: "))
            client_socket.send(str(amount).encode())

        server_response = client_socket.recv(1024).decode()
        print(server_response)

        if choice == 4:
            break

# Close the client socket
client_socket.close()
